sdk/servicebus/azure-servicebus/tests/test_queues.py

```diff
@@ -1271,9 +1271,16 @@ def message_content():
                 sender.send_messages(message)
 
             with sb_client.get_queue_receiver(servicebus_queue.name) as receiver:
-                messages = receiver.receive_messages(max_batch_size=20, max_wait_time=5)
+                receive_counter = 0
+                message_received_cnt = 0
+                while message_received_cnt < 20:
```

Comment:
This makes me nervous.  If something were to go wrong for whatever reason this test would now hang effectively forever, which is not a great failure mode.

I'm assuming the test was flaky because you might have a gap between messages in a given receive batch; I recall seeing a pattern elsewhere such that we call this in a loop until messages == []? (rather than while-forever)